File: services/ts-koa-superstruct/check_service.ts
import * as models from './models'

export type CheckEmptyResponse =
    | { status: "ok" }

export interface CheckQueryParams {
    p_string: string,
    p_string_opt?: string | undefined,
    p_string_array: string[],
    p_date: string,
    p_date_array: string[],
    p_datetime: Date,
    p_int: number,
    p_long: number,
    p_decimal: number,
    p_enum: models.Choice,
    p_string_defaulted: string,
}

export type CheckQueryResponse =
    | { status: "ok" }

export interface CheckUrlParamsParams {
    int_url: number,
    string_url: string,
    float_url: number,
    bool_url: boolean,
    uuid_url: string,
    decimal_url: number,
    date_url: string,
    enum_url: models.Choice,
}

export type CheckUrlParamsResponse =
    | { status: "ok" }

export type CheckForbiddenResponse =
    | { status: "ok", data: models.Message }
    | { status: "forbidden" }

export interface CheckService {
    checkEmpty(): Promise<CheckEmptyResponse>
    checkQuery(params: CheckQueryParams): Promise<CheckQueryResponse>
    checkUrlParams(params: CheckUrlParamsParams): Promise<CheckUrlParamsResponse>
    checkForbidden(): Promise<CheckForbiddenResponse>
}

File: services/go/services/check_service.go
package services

import (
	"./check"
	"./models"
	"cloud.google.com/go/civil"
	"errors"
	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type CheckService struct{}

func (service *CheckService) CheckEmpty() (*check.CheckEmptyResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *CheckService) CheckQuery(pString string, pStringOpt *string, pStringArray []string, pDate civil.Date, pDateArray []civil.Date, pDatetime civil.DateTime, pInt int, pLong int64, pDecimal decimal.Decimal, pEnum models.Choice, pStringDefaulted string) (*check.CheckQueryResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *CheckService) CheckUrlParams(intUrl int64, stringUrl string, floatUrl float32, boolUrl bool, uuidUrl uuid.UUID, decimalUrl decimal.Decimal, dateUrl civil.Date, enumUrl models.Choice) (*check.CheckUrlParamsResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *CheckService) CheckForbidden() (*check.CheckForbiddenResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *CheckService) SameOperationName() (*check.SameOperationNameResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
